op-node/rollup/derive/channel_out_test.go

```diff
@@ -219,42 +230,46 @@ func TestBlockToBatchValidity(t *testing.T) {
 	require.ErrorContains(t, err, "has no transactions")
 }
 
-func SpanChannelAndBatches(t *testing.T, target uint64, len int, algo CompressionAlgo) (*SpanChannelOut, []*SingularBatch) {
+func SpanChannelAndBatches(t *testing.T, targetOutputSize uint64, numBatches int, algo CompressionAlgo, opts ...SpanChannelOutOption) (*SpanChannelOut, []*SingularBatch) {
 	// target is larger than one batch, but smaller than two batches
 	rng := rand.New(rand.NewSource(0x543331))
-	chainID := big.NewInt(rng.Int63n(1000))
+	chainID := rollupCfg.L2ChainID
 	txCount := 1
-	cout, err := NewSpanChannelOut(0, chainID, target, algo, rollup.NewChainSpec(&rollupCfg))
+	genesisTime := rollupCfg.Genesis.L2Time
+	cout, err := NewSpanChannelOut(genesisTime, chainID, targetOutputSize, algo, rollup.NewChainSpec(&rollupCfg), opts...)
 	require.NoError(t, err)
-	batches := make([]*SingularBatch, len)
+	batches := make([]*SingularBatch, 0, numBatches)
 	// adding the first batch should not cause an error
-	for i := 0; i < len; i++ {
+	for i := 0; i < numBatches; i++ {
 		singularBatch := RandomSingularBatch(rng, txCount, chainID)
-		batches[i] = singularBatch
+		// use default 2 sec block time
+		singularBatch.Timestamp = genesisTime + 420_000 + rollupCfg.BlockTime*uint64(i)
+		batches = append(batches, singularBatch)
 	}
 
 	return cout, batches
 }
 
 func TestSpanChannelOut(t *testing.T) {
-	tests := []struct {
-		name string
-		f    func(t *testing.T, algo CompressionAlgo)
-	}{
-		{"SpanChannelOutCompressionOnlyOneBatch", SpanChannelOutCompressionOnlyOneBatch},
-		{"SpanChannelOutCompressionUndo", SpanChannelOutCompressionUndo},
-		{"SpanChannelOutClose", SpanChannelOutClose},
+	tests := []func(t *testing.T, algo CompressionAlgo){
+		SpanChannelOutCompressionOnlyOneBatch,
+		SpanChannelOutCompressionUndo,
+		SpanChannelOutClose,
 	}
 	for _, test := range tests {
 		test := test
 		for _, algo := range CompressionAlgos {
-			t.Run(test.name+"_"+algo.String(), func(t *testing.T) {
-				test.f(t, algo)
+			t.Run(funcName(test)+"_"+algo.String(), func(t *testing.T) {
+				test(t, algo)
 			})
 		}
 	}
 }
 
+func funcName(fn any) string {
+	return runtime.FuncForPC(reflect.ValueOf(fn).Pointer()).Name()
+}
```

Comment:
I don't understand the use of `funcName`. Doesn't the run contain the name of the function already just by how `go test` operates?

Are there any issues with `FuncForPC` in other runtimes?

Reply:
`go test` reads the func name of the top-level test (here `TestSpanChannelOut`), but not of any functions passed to `t.Run`, as these could be anything, also anonymous functions. I just did this to avoid manually writing out function names like it was before, and guarantee correct test names. Before it was
```go
		{"SpanChannelOutCompressionOnlyOneBatch", SpanChannelOutCompressionOnlyOneBatch},
		{"SpanChannelOutCompressionUndo", SpanChannelOutCompressionUndo},
		{"SpanChannelOutClose", SpanChannelOutClose},
```
which IMO is unnecessary if it can be resolved using reflection and runtime information.

Reply:
There's some argument to be made that a `map[string]testFn` is simpler than reflection, but I agree that reflection is the more general (and more cool) solution :D. I'm fine with this

Reply:
Coolness was admittedly the ultimate reason I did it 😎